axlearn/common/trainer.py

```diff
@@ -241,58 +241,9 @@ def __init__(
         self._device_monitor = maybe_instantiate(cfg.device_monitor)
         self._recorder = maybe_instantiate(cfg.recorder)
         self._is_initialized: bool = False
-        self._maybe_record_event(measurement.Event.START_ACCELERATOR_INIT)
-
-        if cfg.model.dtype is None:
-            raise ValueError(f"dtype must be explicitly specified for {self.path()}.model")
-        if cfg.model.param_init is None:
-            cfg.model.param_init = DefaultInitializer.default_config()
-            logging.info(
-                "model.param_init is not specified. Default to DefaultInitializer: %s",
-                cfg.model.param_init,
-            )
-
-        self._per_param_train_dtype = maybe_instantiate(
-            canonicalize_per_param_dtype(cfg.train_dtype)
-        )
-
-        # Create the device mesh.
-        if devices is None:
-            self._step_log(
-                "Devices: global=%s local=%s %s",
-                jax.device_count(),
-                jax.local_device_count(),
-                [device.platform for device in jax.local_devices()],
-            )
-        else:
-            local_devices = [d for d in devices.flatten() if d.process_index == jax.process_index()]
-            self._step_log(
-                "Devices: global=%s local=%s %s",
-                len(devices),
-                len(local_devices),
-                [device.platform for device in local_devices],
-            )
-        self._step_log("Mesh shape: %s", cfg.mesh_shape)
-        devices = (
-            utils.create_device_mesh(mesh_shape=cfg.mesh_shape) if devices is None else devices
-        )
-        mesh = jax.sharding.Mesh(devices, cfg.mesh_axis_names)
-        self._step_log("Global mesh: %s", mesh)
-        self._mesh = mesh
-        self._context_manager: Callable[[], ContextManager] = (
-            maybe_instantiate(cfg.context_manager) or contextlib.nullcontext
-        )
-        xsc_check_policy = None
-        if cfg.xsc_check_policy:
-            if jax.default_backend() != "tpu":
-                # XSC is currently only supported on TPU XLA backend.
-                logging.warning(
-                    "xsc_check_policy was set for non-TPU XLA backend. Running without XSC."
-                )
-            else:
-                xsc_check_policy = maybe_instantiate(cfg.xsc_check_policy)
-        self._xsc_check_policy: Optional[Callable[[int], bool]] = xsc_check_policy
-        self._compiled_train_step: Optional[jax.stages.Compiled] = None
+        # Accelerator initialization.
+        with self._record_event(measurement.Event.ACCELERATOR_INIT):
+            self._device_init(devices)
```

Comment:
Looks like we are downscoping "ACCELERATOR_INIT" here.  I think we have been using "ACCELERATOR_INIT" event for the whole `__init__` call.   Can we keep it that way?

If we want to make it more accurate, I believe `ACCELERATOR_INIT` is only about mesh initialization.  But knowing how long it takes for the whole `__init__` call is probably more valuable

Reply:
You're right! Reverted to the original scope!